starter-code/src/clue.js

```diff
@@ -1,94 +1,205 @@
+        age:          26,
+        image:        "http://www.radiotimes.com/uploads/images/Original/111967.jpg",
+        ocupation:   "Scientist"},
+        {
+        first_name:   "Victor",
+        last_name:    "Plum",
+        color:        "purple",
+        description:  "Billionare video game designer",
+        age:          22,
+        image:        "https://metrouk2.files.wordpress.com/2016/07/professor-plum.jpg",
+        occupation:   "Designer"
+        },
+
+        {first_name:   "Kasandra",
+        last_name:    "Scarlet",
+        color:        "red",
+        description:  "She is an A-list movie star with a dark past",
+        age:          31,
+        image:        "https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg",
+        occupation:   "Actor"
+        },
+
+        {
+        first_name:   "Eleanor",
+        last_name:    "Peacock",
+        color:        "blue",
+        description:  "She is from a wealthy family and uses her status and money to earn popularity",
+        age:          36,
+        image:        "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+        occupation:   "Socialité"   
+        },
+
+        {
+        first_name:   "Jack",
+        last_name:    "Mustard",
+        color:        "yellow",
+        description:  "He is a former football player who tries to get by on his former glory",
+        age:          62,
+        image:        "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+        occupation:   "Retired Football player",  
+        }
+];
+
 
 // Rooms' Collection
-var roomsArray = [];
+var roomsArray = ["Dinning Room",
+    "Conservatory",
+    "Kitchen",
+    "Study",
+    "Library",
+    "Billiard Room",
+    "Lounge",
+    "Ballroom",
+    "Hall",
+    "Spa",
+    "Living Room",
+    "Observatory",
+    "Theater",
+    "Guest House",
+    "Patio"];
 
 // Weapons Collection
-var weaponsArray = [];
+var weaponsArray = [
+    {name: "rope"        ,   weight: 10},
+    {name: "knife"       ,   weight: 8},
+    {name: "candlestick" ,   weight: 2},
+    {name: "dumbbell"  ,   weight: 30},
+    {name: "poison"   ,   weight: 2},
+    {name: "axe"    ,   weight: 15},
+    {name: "bat"     ,   weight: 13},
+    {name: "trophy"      ,   weight: 25},
+    {name: "pistol"      ,   weight: 20}
+];
+
+function randomSelector (arrayEntry) {
+    return arrayEntry[Math.floor(Math.random() * arrayEntry.length)];
+};
+
+
+// var misteryEnvelope;
+
+function pickMistery(){
+    
+    misteryEnvelope = [ randomSelector(charactersArray)  , randomSelector(weaponsArray), randomSelector(roomsArray) ];
+    return  misteryEnvelope;
+    
+};
+
+var misteryEnvelope = pickMistery();
+
+function revealMistery(result){
+result = misteryEnvelope[0].first_name +" "+ misteryEnvelope[0].last_name + " killed Mr.Boddy using the " + misteryEnvelope[1].name + " in the " + misteryEnvelope[2] +"!!!!" ;
+return result;
+}
```

Comment:
one small mistake here : result is what the function returns, so it can't be passed to the function as an argument. You are creating the result variable inside the function, so again you should declare it writing 'var' before. The information the function really needs in order to work is the misteryEnvelope, you should pass this as an argument :)
Right now, when jasmine tests the function,  it tries to pass a certain mistery envelope as an argument but the function currently ignores it and uses the var you created earlier as the array in which it looks for the answer to return, this is why you get an error on Jasmine. Change the parameter and it will be fine !